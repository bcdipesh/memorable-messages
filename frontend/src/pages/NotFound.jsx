import { Link } from "react-router-dom";

import { Button } from "@/components/ui/button";

const NotFound = () => (
  <div className="not-found container flex h-screen flex-col justify-center gap-6">
    <h1 className="text-3xl font-bold">Page not found</h1>
    <p className="text-2xl">
      The requested page couldn't be located. Checkout for any URL misspelling
      or
    </p>
    <Link to="/">
      <Button type="button">Return to the homepage</Button>
    </Link>
  </div>
);

export default NotFound;
