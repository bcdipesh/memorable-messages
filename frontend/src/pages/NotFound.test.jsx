import React from "react";
import { BrowserRouter } from "react-router-dom";

import { render } from "@testing-library/react";

import NotFound from "@/pages/NotFound";

test("Renders without crashing", function () {
  render(
    <BrowserRouter>
      <NotFound />
    </BrowserRouter>,
  );
});

test("Matches snapshot", function () {
  const { asFragment } = render(
    <BrowserRouter>
      <NotFound />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
