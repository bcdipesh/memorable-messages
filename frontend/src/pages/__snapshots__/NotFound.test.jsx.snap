// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="not-found container flex h-screen flex-col justify-center gap-6"
  >
    <h1
      class="text-3xl font-bold"
    >
      Page not found
    </h1>
    <p
      class="text-2xl"
    >
      The requested page couldn't be located. Checkout for any URL misspelling or
    </p>
    <a
      href="/"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
        type="button"
      >
        Return to the homepage
      </button>
    </a>
  </div>
</DocumentFragment>
`;
